addons/binding/org.openhab.binding.chamberlainmyq/src/main/java/org/openhab/binding/chamberlainmyq/internal/discovery/ChamberlainMyQDeviceDiscoveryService.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.chamberlainmyq.internal.discovery;
+
+import static org.openhab.binding.chamberlainmyq.ChamberlainMyQBindingConstants.*;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.chamberlainmyq.config.ChamberlainMyQDeviceConfig;
+import org.openhab.binding.chamberlainmyq.handler.ChamberlainMyQGatewayHandler;
+import org.openhab.binding.chamberlainmyq.handler.ChamberlainMyQGatewayHandler.RequestCallback;
+import org.openhab.binding.chamberlainmyq.internal.ChamberlainMyQHandlerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * Send DiscoveryService Callbacks
+ *
+ * @author Scott Hanson - Initial contribution
+ */
+
+public class ChamberlainMyQDeviceDiscoveryService extends AbstractDiscoveryService {
```

Comment:
Please make this an OSGi `@Component`. Add the class to `getServices()` of the corresponding thing handler and the framework will start the component for you.

Reply:
Sorry, I don't know how to do this, do you know of an example?